src/nvim/cmdexpand.c

```diff
@@ -2531,6 +2541,20 @@ int wildmenu_translate_key(CmdlineInfo *cclp, int key, expand_T *xp, int did_wil
 {
   int c = key;
 
+  if (cmdline_pum_active()) {
+    // When the popup menu is used, Up/Down keys go to the previous and
+    // next items in the menu and Left/Right keys go up/down a directory.
+    if (c == K_UP) {
+      c = K_LEFT;
+    } else if (c == K_DOWN) {
+      c = K_RIGHT;
+    } else if (c == K_LEFT) {
+      c = K_UP;
+    } else if (c == K_RIGHT) {
+      c = K_DOWN;
+    }
+  }
```

Comment:
We have discussed this in an issue long before the vim patch and concluded that we do not want this, at least not by default. key mappings are _semantic_, not defined by some visual property, which also includes arrow keys. (that vim chooses to neglect this fact doesn't mean we also have to jump off the same cliff). For instance: it would break any cnoremap which is based on the actual meaning of the keys.

Reply:
I'll remove this and update `vim_diff.txt` then.